src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -4564,14 +4564,34 @@ public void testRemoveService() throws Exception {
   @Test
   public void testInspectTask() throws Exception {
     requireDockerApiVersionAtLeast("1.24", "swarm support");
+    final Date start = new Date();
 
-    final ServiceSpec spec = createServiceSpec(randomName());
+    final ServiceSpec serviceSpec = createServiceSpec(randomName());
     assertThat(sut.listTasks().size(), is(0));
-    sut.createService(spec);
+    final ServiceCreateResponse serviceCreateResponse = sut.createService(serviceSpec);
     await().until(numberOfTasks(sut), is(greaterThan(0)));
-    final Task task = sut.listTasks().get(0);
-    final Task inspectTask = sut.inspectTask(task.id());
-    assertThat(task, equalTo(inspectTask));
+
+    final Task someTask = sut.listTasks().get(0);
+    final Task inspectedTask = sut.inspectTask(someTask.id());
+    assertThat(inspectedTask.id(), notNullValue());
+    assertThat(inspectedTask.version().index(), allOf(notNullValue(), greaterThan(0L)));
+    final Date now = new Date();
+    assertThat(inspectedTask.createdAt(), allOf(notNullValue(), greaterThan(start), lessThan(now)));
+    assertThat(inspectedTask.updatedAt(), allOf(notNullValue(), greaterThan(start), lessThan(now)));
+    assertThat(inspectedTask.slot(), allOf(notNullValue(), greaterThan(0)));
+    assertThat(inspectedTask.status(), notNullValue());
+    assertThat(inspectedTask.name(), nullValue());
+    assertEquals(serviceCreateResponse.id(), inspectedTask.serviceId());
+    assertEquals(serviceSpec.taskTemplate(), inspectedTask.spec());
+    if (serviceSpec.labels() == null || serviceSpec.labels().isEmpty()) {
```

Comment:
And I want to `assertTrue` on 4589 because I'm saying "I gave you a `ServiceSpec` with either null or empty labels, so the `Task` should also have null or empty labels."